client/change.go

```diff
@@ -0,0 +1,100 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+)
+
+// A Change is a modification to the system state.
+type Change struct {
+	ID      string  `json:"id"`
+	Kind    string  `json:"kind"`
+	Summary string  `json:"summary"`
+	Status  string  `json:"status"`
+	Tasks   []*Task `json:"tasks,omitempty"`
+}
+
+// A Task is an operation done to change the system's state.
+type Task struct {
+	Kind     string       `json:"kind"`
+	Summary  string       `json:"summary"`
+	Status   string       `json:"status"`
+	Log      []string     `json:"log,omitempty"`
+	Progress TaskProgress `json:"progress"`
+}
+
+type TaskProgress struct {
+	Done  int
```

Comment:
`Done => Current`.. I think that's the term we used in the server? I also wondered before if we shouldn't represent that as a sub-document in the style we used in the server instead of an array of two ints. If we add any other metadata it'll look a bit awkward (say, progress to time ready, or time left). WDYT?